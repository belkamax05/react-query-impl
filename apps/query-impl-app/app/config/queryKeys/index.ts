import {
  DeepRecord,
  QueryKeyFunction,
  validateConfigType,
} from '@query-impl/core';
import { QueryKey } from '@tanstack/react-query';

const queryKeys = validateConfigType(
  {
    app: {
      dashboard: ['app', 'dashboard'],
    },
    samples: {
      counter: ['samples', 'counter'],
      todo: ['samples', 'todo'],
    },
  } as const,
  null as DeepRecord<string, QueryKey | QueryKeyFunction>
);

export default queryKeys;
