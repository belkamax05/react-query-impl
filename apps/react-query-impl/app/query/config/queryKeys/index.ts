import { DeepRecord, validateConfigType } from '@rq-impl/core';
import { QueryKey } from '@tanstack/query-core';

type QueryFunction = (...args: unknown[]) => QueryKey;

/**
 * reuseable query keys for react-query, every key should be compatible with QueryKey (string | number | serializable object)[]
 */
const queryKeys = validateConfigType(
  {
    dashboard: ['dashboard'],
    todo: ['todo'],
  } as const,
  null as DeepRecord<string, QueryKey | QueryFunction>
);

export default queryKeys;
