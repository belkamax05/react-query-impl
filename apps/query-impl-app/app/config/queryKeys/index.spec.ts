import queryKeys from '.';

describe('queryKeys', () => {
  it('should match snapshot', () => {
    expect(queryKeys).toMatchSnapshot();
  });
});
