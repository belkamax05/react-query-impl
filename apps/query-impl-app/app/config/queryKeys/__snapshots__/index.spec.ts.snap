// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queryKeys should match snapshot 1`] = `
{
  "app": {
    "dashboard": [
      "app",
      "dashboard",
    ],
  },
  "samples": {
    "counter": [
      "samples",
      "counter",
    ],
    "todo": [
      "samples",
      "todo",
    ],
  },
}
`;
